### PR TITLE
ENG-8840 plan based view catch up

### PR DESCRIPTION
**Pull request by yzhang1991**

### CODE REVIEW

File: src/ee/storage/persistenttable.h [hunk 1]
```diff
@@ -210,6 +212,9 @@ class PersistentTable : public Table, public UndoQuantumReleaseInterest,
   public:
     virtual ~PersistentTable();
 
+    void setViewsUpdateEnabled(bool enabled) { m_viewsUpdateEnabled = enabled; }
+    bool isViewsUpdateEnabled() { return m_viewsUpdateEnabled; }
```
Comment (paulmartel) [inline src/ee/storage/persistenttable.h]: Is there code missing from this change set?
Where's the caller of setViewsUpdateEnabled?

Is this public getter function actually needed?

File: src/frontend/org/voltdb/planner/PlanAssembler.java [hunk 2]
```diff
@@ -303,13 +303,6 @@ void setupForNewPlans(AbstractParsedStmt parsedStmt) {
             return;
         }
 
-        // @TODO
-        // Need to use StmtTableScan instead
-        // check that no modification happens to views
-        if (tableListIncludesView(parsedStmt.m_tableList)) {
-            throw new PlanningErrorException("Illegal to modify a materialized view.");
-        }
-
```
Comment (paulmartel) [inline src/frontend/org/voltdb/planner/PlanAssembler.java]: Was this exception never thrown previously -- is/was an end-user's attempt to update a materialized view caught somewhere else/earlier? Is there a junit test validating that such a guard still exists?

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 3]
```diff
@@ -1971,33 +1972,42 @@ else if (type != ConstraintType.PRIMARY_KEY &&  type != ConstraintType.UNIQUE) {
         catalog_const.setType(type.getValue());
     }
 
+    void compileMatViewQueriesToStatement(Statement stmt, VoltXMLElement xmlquery,
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: Combination of plurals and singulars make the function name hard to understand. How about  compileMatViewQueryToStatement

**Comment by paulmartel** [comment]
Do we have any existing unit tests that populate a table with a view -- we really should.
Then we should add a test case that populates two identical tables, one with view update enabled, one with it disabled, and validates that I can call the buildUp query to bring the two views into synchronization.

File: src/frontend/org/voltdb/compiler/MatViewQueriesXMLGenerator.java [hunk 4]
```diff
@@ -56,6 +67,38 @@ private String lastElementId() {
         return String.valueOf( m_maxElementId );
     }
 
+    private void adjustColumnAliases() {
+        ArrayList<String> realColumnNames = new ArrayList<String>(Collections.nCopies(m_destTable.getColumns().size(), ""));
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/MatViewQueriesXMLGenerator.java]: What's the purpose of these "" values vs. null? Is it really OK for these "" values to survive the first loop and get used as alias values? What does an alias value of "" mean?

**Comment by paulmartel** [comment]
End of review round.

**Comment by yzhang1991** [comment]
Need to remove all the debug outputs.
(This ticket and the ENG-8641 ticket)

File: src/frontend/org/voltdb/planner/PlanAssembler.java (hunk 2 above)
Comment (yzhang1991) [inline src/frontend/org/voltdb/planner/PlanAssembler.java]: We have a guard against modifying materialized view in hsql.

File: src/frontend/org/voltdb/compiler/MatViewQueriesXMLGenerator.java [hunk 5]
```diff
@@ -56,6 +67,38 @@ private String lastElementId() {
...
+        Collections.fill(realColumnNames, null);
+        for (Column info : m_destTable.getColumns()) {
+            realColumnNames.set(info.getIndex(), info.getName());
+        }
+        List<VoltXMLElement> viewColumns = VoltXMLElementHelper.getFirstChild(m_xml, "columns").children;
+        for (int i=0; i<viewColumns.size(); ++i) {
+            VoltXMLElement columnref = viewColumns.get(i);
+            String realColumnName = realColumnNames.get(i);
+            columnref.attributes.put("alias", realColumnName);
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/MatViewQueriesXMLGenerator.java]: add assert( ! "".equals(realColumnName))

**Pull request closed without merging**